File: src/src/bmkim/level_2/기능개발.java
import java.util.*;

public class 기능개발 {
    public static void main(String[] args) {
        int[] progresses = {95, 90, 99, 99, 80, 99};
        int[] speeds = {1, 1, 1, 1, 1, 1};
        solution(progresses, speeds);
    }

    public static int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> queue = new LinkedList<>();

        for (int i=0; i<progresses.length; i++) {
            int completeWorkDay = (int) Math.ceil((double)(100 - progresses[i]) / speeds[i]);
            queue.add(completeWorkDay);
        }

        while (!queue.isEmpty()) {
            Integer start = queue.poll();
            int count = 1;
            while (!queue.isEmpty() && start >= queue.peek()) {
                count++;
                queue.poll();
            }
            answer.add(count);
        }

        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: src/src/hvsun/level_2/기능개발.js
function solution(progresses, speeds) {
    let answer = []
  
    const getCompletedTaskCount = () => {
      let count = 0
      while (progresses.length > 0 && progresses[0] >= 100) {
        progresses.shift();
        speeds.shift();
        count++;
      }
      return count;
    }
  
    while (progresses.length > 0) {
      for (let i = 0; i < progresses.length; i++) {
        progresses[i] += speeds[i]
      }
  
      if (progresses[0] >= 100) {
        const count  = getCompletedTaskCount()
        answer.push(count ? count : 1)
      }
    }
  
    return answer;
  }